csharp/src/Bargreen.Services/InventoryService.cs

```diff
@@ -101,13 +108,67 @@ public IEnumerable<AccountingBalance> GetAccountingBalances()
                      ItemNumber = "fbr77",
                      TotalInventoryValue = 17.99M
                 }
-            };
+                };
+            });
         }
 
-        public static IEnumerable<InventoryReconciliationResult> ReconcileInventoryToAccounting(IEnumerable<InventoryBalance> inventoryBalances, IEnumerable<AccountingBalance> accountingBalances)
+        public async Task<IEnumerable<InventoryReconciliationResult>> ReconcileInventoryToAccounting(IEnumerable<InventoryBalance> inventoryBalances, IEnumerable<AccountingBalance> accountingBalances)
         {
             //TODO-CHALLENGE: Compare inventory balances to accounting balances and find differences
-            throw new NotImplementedException();
+
+            //insntantiate list of InventoryRecResults
+            List<InventoryReconciliationResult> inventoryReconciliationResults = new List<InventoryReconciliationResult>();
+           
+            // iterate through both the inventtory and accounting balances
+            foreach(InventoryBalance InvItem in inventoryBalances)
+            {
+                foreach(AccountingBalance AccNum in accountingBalances)
+                {
+                    // check that the amounts of items are more than 0
+                    if(InvItem.QuantityOnHand > 0 && AccNum.TotalInventoryValue > 0)
+                    {
+                        //if the items numbers do not match add two reulsts to the list - eahc with their correspoding item number
+                        if (InvItem.ItemNumber != AccNum.ItemNumber)

```

Comment:
If our goal is to identify items that have mismatched balances between the 2 systems, then we don't really care about comparing items that match this condition. You're going to end up with a whole lot of false positives. This makes me think that the problem we're trying to solve isn't very clear.